core/src/main/java/com/linecorp/armeria/client/retry/RetryingHttpClient.java

```diff
@@ -74,15 +83,18 @@
     RetryingHttpClient(Client<HttpRequest, HttpResponse> delegate,
                        RetryStrategy<HttpRequest, HttpResponse> strategy,
                        Supplier<? extends Backoff> backoffSupplier,
-                       int defaultMaxAttempts, boolean useRetryAfter) {
+                       int defaultMaxAttempts, boolean useRetryAfter, int contentPreviewLength) {
         super(delegate, strategy, backoffSupplier, defaultMaxAttempts);
         this.useRetryAfter = useRetryAfter;
+        checkArgument(contentPreviewLength >= 0,
+                      "contentPreviewLength: " + contentPreviewLength + " (expected: >= 0)");
```

Comment:
`"contentPreviewLength: %s (expected: >=0)", contentPreviewLength` (because otherwise we will do string concat even on a valid argument.